giraffe/src/utils/useBandHoverColumns.ts

```diff
@@ -0,0 +1,65 @@
+import {BandIndexMap, LineData, NumericColumnData} from '../types'
+import {useLazyMemo} from './useLazyMemo'
+import {isDefined} from './isDefined'
+
+const buildColumnData = (
+  lineData: LineData,
+  groupColData: NumericColumnData,
+  bandIndexMap: BandIndexMap
+) => {
+  const hoverableColumnData = {
+    xs: [],
+    ys: [],
+    groupColData: [],
+  }
+  const {rowIndices} = bandIndexMap
+
+  rowIndices.forEach(rowIndex => {
+    if (isDefined(rowIndex)) {
+      hoverableColumnData.xs = hoverableColumnData.xs.concat(
+        lineData[rowIndex].xs
+      )
+      hoverableColumnData.ys = hoverableColumnData.ys.concat(
+        lineData[rowIndex].ys
+      )
+      groupColData
+        .filter(index => index === rowIndex)
+        .forEach(value => hoverableColumnData.groupColData.push(value))
```

Comment:
i think we should prefer `map` and returning the array, rather than pushing to another array in these cases. this is what `map` is specifically designed for.

```ts
hoverableColumnData.groupColData = groupColData
  .filter(index => index === rowIndex)
  .map(value => value)
```

what do you think?

Reply:
Yes, normally we would prefer map over forEach. However, in this piece of code after we map, the results have to somehow combine with another array. concat can take care of that for normal arrays. groupColData is a `Float64Array` cannot be an argument to concat.

The other option is to use the spread operator to spread out groupColData's elements into the concat function. But, there could be _many_ elements which causes problems with function calls with too many elements.

So, i think here because of the `Float64Array`, pushing them to an array is as good as it gets.

Reply:
@TCL735 can you add a comment explaining those considerations?

Reply:
I ended up doing the destructuring into a literal array, which is backed by iterable objects and should be able to handle as many elements as any array can hold. So, no comment needed here.